website/blog/2024-12-24-the-rise-and-evolution-of-rag-in-2024-a-year-in-review/index.md

```diff
@@ -0,0 +1,296 @@
+
+However, cross-encoders also have their drawbacks: for encoders, document embeddings can be completed during the offline indexing phase, allowing for quick retrieval by encoding only the query during online queries. In contrast, cross-encoders require cross-encoding and model output for each query-document pair, which incurs high computational costs. Therefore, they are suitable only for re-ranking, and there cannot be too many coarse filtering results; otherwise, it will significantly increase query latency.
+
+![](./cross_encoder.jpg)
+
+When evaluating embedding models and reranker models, the MTEB leaderboard is often referenced. In the first half of 2024, the reranking leaderboard was primarily dominated by various cross-encoders, while in the second half, it was increasingly occupied by reranking models based on large language models (LLMs). For instance, the current top-ranking model, gte-Qwen2-7B [Reference 31], is fine-tuned from the Qwen2 7B base model. This approach no longer uses a traditional encoder architecture but instead employs a standard LLM decoder architecture, resulting in a larger parameter count and higher reasoning costs.
+
+Considering both ranking effectiveness and cost, a reranking solution known as the delayed interaction model has gained attention; this involves tensor-based reranking, as illustrated in the diagram below.
+
+![](./reranking_delayed_interaction.jpg)
+
+The specific approach is as follows: during the indexing phase, embeddings generated by the encoder for each token are stored. Thus, a document can be represented by a tensor (or multiple vectors). During querying, embeddings for each token in the query are generated, and the pairwise similarity between all tokens in the query and the text chunks is calculated. The final document score is obtained by summing these similarities. This reranking method also captures the interaction information between tokens, allowing it to theoretically achieve performance comparable to that of a cross-encoder.
+
+On the other hand, because complex model reasoning is not involved during querying, the cost is significantly lower than that of cross-encoders or LLM-based rerankers. This can even enable sorting to be performed within the database itself. The benefits include the ability to rerank more results, which increases the likelihood of compensating for previous recall shortcomings, even if the initial filtering results are not ideal. The following figure compares the evaluation results of applying tensor reranking based on single-recall, dual-recall, and triple-recall using the Infinity database.
```

Comment:
complex model reasoning => complex model inference

This can even enable sorting to be performed within the database itself. =>
This can even enable ranking to be performed within the database itself.